applications/StructuralMechanicsApplication/custom_strategies/custom_strategies/prebuckling_strategy.hpp

```diff
@@ -0,0 +1,980 @@
+        {
+            //#####################################################################################
+            //Method 1
+            //Poy, Denise Lori-eng; On the Buckling Finite Element Analysis of Beam Structures 
+            //#####################################################################################
+            // Solve Eigenvalue Problem
+            // BuiltinTimer eigen_solve_time;
+            // this->pGetEigenSolver()->GetLinearSystemSolver()->Solve(
+            //     rStiffnessMatrix,
+            //     rStiffnessMatrixPrevious,
+            //     Eigenvalues,
+            //     Eigenvectors);
+
+            // lambda = 1.0 / ( 1.0 - Eigenvalues(0) ) * deltaLoadMultiplier;
+            // for(int i = 0; i < Eigenvalues.size(); i++ )
+            // {
+            //     Eigenvalues[i] = lambdaPrev + 1.0 / ( 1.0 - Eigenvalues[i] )*deltaLoadMultiplier;
+            // }
+            // this->AssignVariables(Eigenvalues, Eigenvectors);
+            //End Method 1#########################################################################
+
+            //#####################################################################################
+            //Method 2: 
+            // Jia, X.Mang, H. A.; Assessment of solutions from the consistently linearized eigenproblem 
+            // by means of finite difference approximations
+            //#####################################################################################
+            // Solve Eigenvalue Problem
+
+            // This implementation of the eigenvalue problem relies on the difference of the stiffness matrix 
+            // between the current and previous step
+            rStiffnessMatrix = rStiffnessMatrixPrevious - rStiffnessMatrix;
+
+            this->pGetEigenSolver()->GetLinearSystemSolver()->Solve(
+                rStiffnessMatrixPrevious,
+                rStiffnessMatrix,
+                Eigenvalues,
+                Eigenvectors);
+
+            // Update eigenvalues to loadfactors (Instead of dividing matrix by deltaLoadMultiplier, here eigenvalues are multiplied)
+            lambda = Eigenvalues(0)*deltaLoadMultiplier;
+            for(int i = 0; i < Eigenvalues.size(); i++ )
+            {
+                Eigenvalues[i] = lambdaPrev + Eigenvalues[i]*deltaLoadMultiplier;
+            }
+            
+            this->AssignVariables(Eigenvalues, Eigenvectors);
+            
+            // Reset elastic stiffness matrix for next loadstep
+            this->pGetStiffnessMatrix() = nullptr;
+            pBuilderAndSolver->ResizeAndInitializeVectors(
+                pScheme, pGetStiffnessMatrix(), pGetDx(), pGetRHS(), rModelPart);
+
+            //End Method 2#########################################################################
+
+            // Convergence criteria for buckling analysis
+            if( abs(lambda/lambdaPrev) < mConvergenceRatio )
```

Comment:
![imagen](https://user-images.githubusercontent.com/19991680/35767644-8a9c6856-08f0-11e8-9944-e72c6c1db7b3.png)

Use std::abs, otherwise will take an integer, otherwise will converge always that is minor of 1.0
```suggestion
            if( std::abs(lambda/lambdaPrev) < mConvergenceRatio )
```

Reply:
abs changed